File: mks/fresh/pActionsMenuBar.cpp
'''***************************************************************************
**
** 		Created using Monkey Studio IDE v1.8.4.0 (1.8.4.0)
** Authors   : Filipe AZEVEDO aka Nox P@sNox <pasnox@gmail.com>
** Project   : Fresh Library
** FileName  : pActionsMenuBar.cpp
** Date      : 2011-02-20T00:41:09
** License   : LGPL v3
** Home Page : https:#github.com/pasnox/fresh
** Comment   : Fresh Library is a Qt 4 extension library providing set of core & gui classes.
**
** This program is free software: you can redistribute it and/or modify
** it under the terms of the GNU Leser General Public License as published by
** the Free Software Foundation, version 3 of the License, or
** (at your option) any later version.
**
** This package is distributed in the hope that it will be useful,
** but WITHOUT ANY WARRANTY; without even the implied warranty of
** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
** GNU Lesser General Public License for more details.
**
** You should have received a copy of the GNU Lesser General Public License
** along with self program. If not, see <http:#www.gnu.org/licenses/>.
**
***************************************************************************'''
#include "pActionsMenuBar.h"
#include "pActionsModel.h"

pActionsMenuBar.pActionsMenuBar( QWidget* parent )
	: QMenuBar( parent )
	mModel = 0


def setModel(self, model ):
	if  mModel :		dismModel.actionInserted.connect(self.model_actionInserted)
		dismModel.actionsCleared.connect(self.model_actionsCleared)
		clear()
		mModel = 0

	
	mModel = model
	
	if  mModel :		for ( i = 0; i < mModel.rowCount(); i++ )			action = mModel.action( mModel.index( i, 0 ) )
			model_actionInserted( action )


	
	if  mModel :		mModel.actionInserted.connect(self.model_actionInserted)
		mModel.actionsCleared.connect(self.model_actionsCleared)



def model(self):
	if  not mModel :		mb = const_cast<pActionsMenuBar*>( self )
		mb.setModel( pActionsModel( mb ) )

	
	return mModel


def model_actionInserted(self, action ):
	parent = mModel.parent( action )
	
	if  not parent and action.menu() :		addMenu( action.menu() )



def model_actionsCleared(self):
	clear()

File: mks/fresh/actionmanager/pActionsMenuBar.py
"""This file has been ported from fresh library by Azevedo Filippe aka PasNox

See information at https://github.com/pasnox/fresh and 
API docks at http://api.monkeystudio.org/fresh/
"""

from PyQt4.QtGui import QMenuBar

from .ActionManager import pActionsManager

class pActionsMenuBar(QMenuBar):
    def __init__(self, parent):
        QMenuBar.__init__(self, parent)
        self._manager = None

    def setModel(self, model ):
        if self._manager is not None:
            self._manager.actionInserted.disconnect(self.model_actionInserted)
            self.clear()
            self._manager = None
        
        self._manager = model
        
        if  self._manager is not None:
            for action in self._manager.children(None):
                self.model_actionInserted( action )

        if self._manager is not None:
            self._manager.actionInserted.connect(self.model_actionInserted)

    def model(self):
        if self._manager is None:
            self.setModel( pActionsManager(self) )

        return self._manager

    def model_actionInserted(self, action ):
        parent = self._manager.parentAction( action )
        
        if parent is None and action.menu():
            self.addMenu( action.menu() )
